### PR TITLE
REDDEE-1835 Implement a dialog for filtering Java classes

### PR DESCRIPTION
**Pull request by luvalach**
REDDEE-1835 Implement a dialog for filtering Java classes

Signed-off-by: Lukáš Valach <lvalach@redhat.com>

### CODE REVIEW

**Comment by genie-reddeer** [comment]
Can one of the admins verify this patch?

**Comment by genie-reddeer** [comment]
Can one of the admins verify this patch?

**Comment by rawagner** [comment]
testPR

**Comment by luvalach** [comment]
One of my tests failed, I have to fix it.

**Comment by luvalach** [comment]
Could anybody run the tests again please?

**Comment by odockal** [comment]
testPR

**Comment by luvalach** [comment]
One test is still failing but unfortunately I can't reproduce that problem locally. Hope it is fixed now. @odockal , could you try it again please?

**Comment by odockal** [comment]
testPR

**Comment by luvalach** [comment]
Great, it passed finally.

**Comment by luvalach** [comment]
Good point, thank you @odockal. Could you check it again please? And also run the tests again. It passed locally but one never know.

**Comment by rawagner** [comment]
testPR

File: plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/jdt/ui/dialogs/FilteredTypesSelectionDialog.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * Represents {@value #DIALOG_TITLE} dialog.
+ * 
+ * @see org.eclipse.jdt.internal.ui.dialogs.FilteredTypesSelectionDialog
+ * 
+ * @author lvalach
+ *
+ */
+public class FilteredTypesSelectionDialog extends FilteredItemsSelectionDialog {
```
[the remark this answers is not in the document]
Comment (rawagner) [reply plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/jdt/ui/dialogs/FilteredTypesSelectionDialog.java]: @odockal -1 from me. The reasons are
1. The other dialogs can change anytime
2. If I open for example Open Task dialog, hit ALT+SHIFT+F1 to find out the name of the class I will not be able to find it in RedDeer
3.  abstract FilteredItemsSelectionDialog can be used if the concrete implementation is missing

**Comment by rawagner** [comment]
@luvalach can you please rename FilteredTypesSelectionDialog to OpenTypeSelectionDialog ? Thx!

**Comment by luvalach** [comment]
Done. Thank you both for review.

**Pull request merged**